2.1-segwit-version-1.ipynb

```diff
@@ -278,7 +288,136 @@
     "privkey1, pubkey1 = generate_key_pair()\n",
     "privkey2, pubkey2 = generate_key_pair()\n",
     "\n",
-    "# Generate a 2-of-2 aggregate MuSig key\n",
+    "# Create the spending script\n",
+    "multisig_script = CScript([CScriptOp(OP_2), pubkey1.get_bytes(), pubkey2.get_bytes(), CScriptOp(OP_2), CScriptOp(OP_CHECKMULTISIG)])\n",
+    "\n",
+    "# Hash the spending script\n",
+    "script_hash = sha256(multisig_script)\n",
+    "\n",
+    "# Generate the address\n",
+    "version = 0\n",
+    "address = program_to_witness(version, script_hash)\n",
+    "print(\"bech32 address is {}\".format(address))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Example 2.1.8: Start a Bitcoind node and send funds to the segwit v0 address\n",
+    "\n",
+    "We'll use the `generate_and_send_coins()` function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test = util.TestWrapper()\n",
+    "test.setup()\n",
+    "node = test.nodes[0]\n",
+    "\n",
+    "# Generate coins and create an output\n",
+    "tx = node.generate_and_send_coins(address)\n",
+    "print(\"Transaction {}, output 0\\nsent to {}\\n\".format(tx.hash, address))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Example 2.1.9 : Construct CTransaction, sign and check validity\n",
+    "\n",
+    "In this example we:\n",
+    "- create a `CTransaction` object\n",
+    "- create signatures for both public keys\n",
+    "- create a valid witness using those signatures and add it to the transaction\n",
+    "- test transaction validity"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a spending transaction\n",
+    "spending_tx = test.create_spending_transaction(tx.hash)\n",
+    "\n",
+    "# Generate the segwit v0 signature hash for signing\n",
+    "sighash = SegwitV0SignatureHash(script=multisig_script,\n",
+    "                                txTo=spending_tx,\n",
+    "                                inIdx=0,\n",
+    "                                hashtype=SIGHASH_ALL,\n",
+    "                                amount=100_000_000)\n",
+    "\n",
+    "# Sign using ECDSA and append the SIGHASH byte\n",
+    "sig1 = privkey1.sign_ecdsa(sighash) + chr(SIGHASH_ALL).encode('latin-1')\n",
+    "sig2 = privkey2.sign_ecdsa(sighash) + chr(SIGHASH_ALL).encode('latin-1')\n",
+    "\n",
+    "print(\"Signatures:\\n- {},\\n- {}\\n\".format(sig1.hex(), sig2.hex()))\n",
+    "\n",
+    "# Construct transaction witness. For multisig P2WSH, the witness is the signatures and the scipt\n",
+    "witness = CScriptWitness()\n",
+    "witness.stack = [b'', sig1, sig2, multisig_script]\n",
+    "witness_in = CTxInWitness()\n",
```

Comment:
I think witness.stack should be passed to the constructor.

Reply:
as above, caused by bad rebase